### DISCUSSION TITLE
How to open Olympus VSI file

### DISCUSSION THREAD

**Discussion by yq1985** [libvips/libvips#2887] [Help wanted]
Hello,

I was trying to open Olympus VSI via pyvips. But I am only mange to get label+overiew image. Here is my code. Can help to advise what shall be done?

`
src = "image.vsi"

image = pyvips.Image.new_from_file(src, access="sequential",page=3)

for field in image.get_fields():
    try:
        print(field,image.get(field))
    except Exception as inst:
        print("An exception occurred",inst, field)

mem_img = image.write_to_memory()

imgnp=np.frombuffer(mem_img, dtype=np.uint8).reshape(image.height, image.width, 3)

plt.imshow(imgnp)
plt.show()
`

**Comment by jcupitt** [comment]
Hello @yq1985,

openslide does not support Olympus by default. There's some discussion (and a link to a fork) here: https://github.com/openslide/openslide/issues/143
